File: Easy/88. Merge Sorted Array.py
class Solution:
    def merge(self, nums1: list[int], m: int, nums2: list[int], n: int) -> None:
        index1 = 0
        index2 = 0
        mergeList: list[int] = []
        while index1 < m and index2 < n:
            if nums1[index1] < nums2[index2]:
                mergeList.append(nums1[index1])
                index1 += 1
            else:
                mergeList.append(nums2[index2])
                index2 += 1

        while index1 < m:
            mergeList.append(nums1[index1])
            index1 += 1

        while index2 < n:
            mergeList.append(nums2[index2])
            index2 += 1

        for i in range(len(mergeList)):
            nums1[i] = mergeList[i]

File: Easy/88. Merge Sorted Array.cs
public class Solution
{
    public void Merge(int[] nums1, int m, int[] nums2, int n)
    {
        int i = m - 1, j = n - 1;
        int r = nums1.Length - 1;
        while (i >= 0 && j >= 0)
        {
            if (nums1[i] > nums2[j])
            {
                nums1[r] = nums1[i];
                i--;
            }
            else
            {
                nums1[r] = nums2[j];
                j--;
            }
            r--;
        }

        while (i >= 0)
        {
            nums1[r] = nums1[i];
            i--;
            r--;
        }

        while (j >= 0)
        {
            nums1[r] = nums2[j];
            j--;
            r--;
        }
    }
}